tests/thread_tests/thread_test.go

```diff
@@ -0,0 +1,36 @@
+package thread_tests
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/canonical/matter-snap-testing/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAllClustersAppThread(t *testing.T) {
+	setup(t)
+
+	trimmedActiveDataset := getActiveDataset(t)
+
+	remote_setup(t)
+
+	t.Run("Commission", func(t *testing.T) {
+		stdout, _, _ := utils.Exec(t, "sudo chip-tool pairing code-thread 110 hex:"+trimmedActiveDataset+" 34970112332 2>&1")
+		assert.NoError(t,
+			os.WriteFile("chip-tool-thread-pairing.log", []byte(stdout), 0644),
+		)
+	})
+
+	t.Run("Control", func(t *testing.T) {
+		start := time.Now()
+		stdout, _, _ := utils.Exec(t, "sudo chip-tool onoff toggle 110 1 2>&1")
+		assert.NoError(t,
+			os.WriteFile("chip-tool-thread-onoff.log", []byte(stdout), 0644),
+		)
+
+		remote_waitForLogMessage(t, matterGPIOSnap, "CHIP:ZCL: Toggle ep1 on/off", start)
+	})
```

Comment:
It would be good if the tests clean up the environment by toggling again or executing `chip-tool onoff off` so that the GPIO is set back to LOW.